examples/cleanrl_examples/ppo_atari_envpool.py

```diff
@@ -220,25 +212,37 @@ def __init__(self, env, deque_size=100):
     self.num_envs = getattr(env, "num_envs", 1)
     self.episode_returns = None
     self.episode_lengths = None
-    self.is_vector_env = True
+    # get if the env has lives
+    self.has_lives = False
+    env.reset()
+    info = env.step(np.zeros(self.num_envs, dtype=int))[-1]
+    if info["lives"].sum() > 0:
+      self.has_lives = True
+      print("env has lives")
 
   def reset(self, **kwargs):
     observations = super(RecordEpisodeStatistics, self).reset(**kwargs)
     self.episode_returns = np.zeros(self.num_envs, dtype=np.float32)
     self.episode_lengths = np.zeros(self.num_envs, dtype=np.int32)
+    self.lives = np.zeros(self.num_envs, dtype=np.int32)
     self.returned_episode_returns = np.zeros(self.num_envs, dtype=np.float32)
     self.returned_episode_lengths = np.zeros(self.num_envs, dtype=np.int32)
     return observations
 
   def step(self, action):
     observations, rewards, dones, infos = super(RecordEpisodeStatistics,
                                                 self).step(action)
-    self.episode_returns += rewards
+    self.episode_returns += infos["reward"]
     self.episode_lengths += 1
     self.returned_episode_returns[:] = self.episode_returns
     self.returned_episode_lengths[:] = self.episode_lengths
-    self.episode_returns *= (1 - dones)
-    self.episode_lengths *= (1 - dones)
+    all_lives_exhausted = infos["lives"] == 0
+    if self.has_lives:
```

Comment:
@vwxyzjn does this change have anything to do with gym versioning (or termination)? Is this code doing the same thing as
```
        self.episode_returns *= 1 - infos["terminated"]
        self.episode_lengths *= 1 - infos["terminated"]
```
as in [cleanrl repo's example](https://github.com/vwxyzjn/cleanrl/blob/f62ad1f942890db7b752f4ad79b65be53917d3c2/cleanrl/ppo_atari_envpool.py#L103C1-L104C56)?

Why the RecordEpisodeStatistics wrapper is different in 
https://github.com/vwxyzjn/cleanrl/blob/master/cleanrl/ppo_atari_envpool.py
and
https://github.com/sail-sg/envpool/blob/main/examples/cleanrl_examples/ppo_atari_envpool.py
is not clear to me.

Reply:
No you should be doing 

```
  self.episode_returns *= 1 - infos["terminated"]
  self.episode_lengths *= 1 - infos["terminated"]
```